### PR TITLE
Add a method to mark the app as opened

### PR DESCRIPTION
**Pull request by kul3r4**
#### WHAT
Add a method to mark the app has been opened

#### WHY
Apps can track if there is a friction with an initial screen such as the login

#### HOW
track the first time the app is opened

#### Checklist :clipboard:
- [ ] Add explicit visibility modifier and explicit return types for public declarations
- [ ] Run spotless check
- [ ] Run tests
- [ ] Update metalava's signature text files

### CODE REVIEW

File: datalayer-watch/src/main/java/com/google/android/horologist/datalayer/watch/WearDataLayerAppHelper.kt [hunk 1]
```diff
@@ -111,6 +111,19 @@ public class WearDataLayerAppHelper(
         }
     }
 
+    /**
+     * Marks that the app has been opened at least once.
+     */
+    public suspend fun markAppHasBeenOpened() {
```
Comment (yschimke) [inline datalayer-watch/src/main/java/com/google/android/horologist/datalayer/watch/WearDataLayerAppHelper.kt:117]: In terms of naming, what are we trying to convey?

app vs activity?
open vs launch?

I'm ok with it as is, but poking a discussion.

**Review by yschimke — APPROVED** [review]
LGTM

File: datalayer-watch/src/main/java/com/google/android/horologist/datalayer/watch/WearDataLayerAppHelper.kt (hunk 1 above)
Comment (kul3r4) [reply datalayer-watch/src/main/java/com/google/android/horologist/datalayer/watch/WearDataLayerAppHelper.kt:117]: Yep launched makes more sense, will fix

File: datalayer/src/main/proto/app_helper_pb.proto [hunk 2]
```diff
@@ -57,4 +57,5 @@ message SurfacesInfo {
   reserved 1;
   repeated ComplicationInfo complications = 2;
   repeated TileInfo tiles = 3;
+  bool appHasBeenOpened = 4;
```
[the remark this answers is not in the document]
Comment (kul3r4) [reply datalayer/src/main/proto/app_helper_pb.proto]: done

File: datalayer-watch/src/main/java/com/google/android/horologist/datalayer/watch/WearDataLayerAppHelper.kt (hunk 1 above)
Comment (kul3r4) [reply datalayer-watch/src/main/java/com/google/android/horologist/datalayer/watch/WearDataLayerAppHelper.kt]: done

File: datalayer/src/main/proto/app_helper_pb.proto [hunk 3]
```diff
@@ -53,9 +53,15 @@ message TileInfo {
   .google.protobuf.Timestamp timestamp = 3;
 }
 
+message ActivityLaunched {
+  reserved 1;
```
Comment (yschimke) [inline datalayer/src/main/proto/app_helper_pb.proto:57]: nit: you shouldn't need reserved as part of the initial checkin, really only if we released it during this time.

File: datalayer-watch/src/main/java/com/google/android/horologist/datalayer/watch/WearDataLayerAppHelper.kt [hunk 4]
```diff
@@ -112,13 +113,13 @@ public class WearDataLayerAppHelper(
...
+     * Marks that the main activity has been launched at least once.
      */
-    public suspend fun markAppHasBeenOpened() {
+    public suspend fun markActivityLaunchedOnce() {
         surfaceInfoDataStore.updateData { info ->
             info.copy {
-                if (!appHasBeenOpened) {
-                    appHasBeenOpened = true
+                if (!activityLaunched.activityLaunchedOnce) {
+                    activityLaunched = ActivityLaunched.newBuilder().setActivityLaunchedOnce(true).build()
```
Comment (yschimke) [inline datalayer-watch/src/main/java/com/google/android/horologist/datalayer/watch/WearDataLayerAppHelper.kt:122]: Not setting the timestamp here.

Comment (yschimke) [reply datalayer-watch/src/main/java/com/google/android/horologist/datalayer/watch/WearDataLayerAppHelper.kt:122]: Should we update timestamp on each launch? Or do we want it to be time of the first launch?

**Pull request merged**